### PR TITLE
feat(current-progress): allow users to open the credit dialog

### PR DESCRIPTION
**Pull request by wescopeland**
> ✅ Safe to merge, supports the "beat" feature flag.

This PR adds a trigger to open the beaten game credit dialog to the game progress component.

If the game is both beatable _and_ unbeaten, an info circle appears which triggers the dialog when clicked, just like the achievement type indicators.

https://github.com/RetroAchievements/RAWeb/assets/3984985/2e9da98c-786a-4e33-a2f1-1339bcd66cdb

### CODE REVIEW

**Review by Tsearo — APPROVED** [review]
Tested various games including those with past masteries but then had new progression added and the dialog looks to be showing correctly in each case.

File: resources/views/platform/components/game/current-progress/big-status-label.blade.php [hunk 1]
```diff
@@ -35,15 +37,35 @@
...
 
-<p class="text-lg {{ $colorClassName }} mb-2.5 mt-0.5">
-    {{ $statusLabel }}
-</p>
\ No newline at end of file
+<div class="text-lg {{ $colorClassName }} mb-1.5 mt-0.5 flex items-center gap-x-1">
+    <p>{{ $statusLabel }}</p>
+
+    @hasfeature('beat')
+        @if (!$isBeaten)
```
Comment (Jamiras) [inline resources/views/platform/components/game/current-progress/big-status-label.blade.php:55]: Should this also be tied to `$isBeatable`?
image

File: resources/views/platform/components/game/current-progress/big-status-label.blade.php [hunk 2]
```diff
@@ -35,15 +37,35 @@
...
 }
 
+$isBeaten = $isBeatable && ($isBeatenHardcore || $isBeatenSoftcore);
+
 // This case can occur on legacy completion/mastery awards
 // where progression achievements were added after the user
 // had already mastered the game. It's an edge case, but we
 // try to gracefully handle it anyway.
-if (($isCompleted || $isMastered) && ($isBeatable && !$isBeatenHardcore && !$isBeatenSoftcore)) {
+if (($isCompleted || $isMastered) && !$isBeaten) {
```
Comment (Jamiras) [inline resources/views/platform/components/game/current-progress/big-status-label.blade.php:46]: Not sure if it's this line, but something in this PR marks a game as Unbeaten if it's been mastered but there aren't any progression achievements.
image

File: resources/views/platform/components/game/current-progress/big-status-label.blade.php (hunk 1 above)
Comment (wescopeland) [reply resources/views/platform/components/game/current-progress/big-status-label.blade.php]: Yep, resolved in https://github.com/RetroAchievements/RAWeb/pull/1859/commits/bfbca79c9b38b3e3947c8f6a41d20e43b580fb58.

File: resources/views/platform/components/game/current-progress/big-status-label.blade.php (hunk 2 above)
Comment (wescopeland) [reply resources/views/platform/components/game/current-progress/big-status-label.blade.php]: Good catch, fixed in https://github.com/RetroAchievements/RAWeb/pull/1859/commits/bfbca79c9b38b3e3947c8f6a41d20e43b580fb58.

**Review by Jamiras — APPROVED** [review]

**Pull request merged**